src/test/kotlin/com/terraformation/backend/documentproducer/DocumentUpgradeCalculatorTest.kt

```diff
@@ -224,44 +225,64 @@ class DocumentUpgradeCalculatorTest : DatabaseTest(), RunsAsUser {
 
   @Test
   fun `updates variable references in sections to use new variable IDs`() {
-    val oldManifestId = insertVariableManifest()
-    val outdatedVariableId =
-        insertVariableManifestEntry(insertTextVariable(), manifestId = oldManifestId)
-    val unmodifiedVariableId =
-        insertVariableManifestEntry(insertNumberVariable(), manifestId = oldManifestId)
-    val obsoleteVariableId =
-        insertVariableManifestEntry(insertTextVariable(), manifestId = oldManifestId)
-    val sectionVariableId = insertVariableManifestEntry(insertSectionVariable())
-    val newManifestId = insertVariableManifest()
-    val updatedVariableId =
-        insertVariableManifestEntry(
-            insertTextVariable(
-                insertVariable(type = VariableType.Text, replacesVariableId = outdatedVariableId)),
-            manifestId = newManifestId)
```

Comment:
There are a few things in this test that were inconsistent with the way things work now, with separated document template manifests and variables. We no longer have non-section type variables directly attached to variable manifests.

This entire test was rewritten into a more representative flow of how things occur in the system, and one that represents the flow that produced the bug that caused this investigation.